frontend/src/pages/device/index.vue

```diff
@@ -259,7 +307,74 @@ export default {
             this.device = await deviceApi.updateDevice(device.id, { application: applicationId, instance: null })
 
             Alerts.emit('Device successfully assigned to application.', 'confirmation')
+        },
+        openEditor () {
+            window.open(this.deviceEditorURL, `device-editor-${this.device.id}`)
+        },
+        async openTunnel (launchEditor = false) {
+            if (this.device.status === 'running') {
+                if (this.device.editor?.enabled && this.device.editor?.connected) {
+                    this.openEditor()
+                } else {
+                    this.openingTunnel = true
+                    this.$refs.dialog.show()
+                    try {
+                        // * Enable Device Editor (Step 1) - (browser->frontendApi) User clicks button to "Enable Editor"
+                        const result = await deviceApi.enableEditorTunnel(this.device.id)
+                        this.updateTunnelStatus(result)
+                        setTimeout(() => {
+                            this.loadDevice()
+                            if (launchEditor && this.device.editor?.enabled && this.device.editor?.connected) {
+                                this.openEditor()
+                            }
+                            this.$refs.dialog.close()
+                        }, 500)
+                    } finally {
+                        this.openingTunnel = false
+                    }
+                }
+            } else {
+                Alerts.emit('The device must be in "running" state to access the editor', 'warning', 7500)
```

Comment:
Hmmm. Not sure TBH. 

It could be any number of reasons - far more than would "look nice" in a toast.

For example: The device might be off-line due to network, power, busy starting up, removed, and so on. 

How about:

```suggestion
                Alerts.emit('Unable to establish a connection to the device. Please ensure it is powered on, connected to the network and is in the "running" state before trying again', 'warning', 7500)
```

Reply:
@knolleary do you have a cleaner, more succinct suggestion pls? ^

Reply:
Something like:

> Unable to establish a connection to the device. Please check it is connected and try again

Reply:
```suggestion
                Alerts.emit('Unable to establish a connection to the device. Please check it is connected and running then try again', 'warning', 7500)
```

Slight variation for consideration.  Since the logic test that leads to this toast is that the device status is NOT "running" (i.e. it could be starting up or crashed or some other state).

Thoughts please @knolleary